src/cc_helpers.cc

```diff
@@ -2383,24 +2390,26 @@ certifier::framework::secure_authenticated_channel::
 bool certifier::framework::secure_authenticated_channel::init_client_ssl(
     const string &host_name,
     int           port,
-    string &      asn1_root_cert,
+    const string &asn1_root_cert,
     key_message & private_key,
     const string &auth_cert) {
 
   OPENSSL_init_ssl(0, NULL);
   SSL_load_error_strings();
 
   private_key_.CopyFrom(private_key);
+
   asn1_root_cert_.assign((char *)asn1_root_cert.data(), asn1_root_cert.size());
+
   root_cert_ = X509_new();
-  if (!asn1_to_x509(asn1_root_cert, root_cert_)) {
+  if (!asn1_to_x509(asn1_root_cert_, root_cert_)) {
```

Comment:
I think this change and the change on L2409, L2412 should be ok, as we are copying-in the input `asn1_root_cert` to this member field on L2402, above.

Do give this a good look and confirm that I'm not making a mistake here ...

Reply:
Was this a bug before the change? It seems that d2i_X509 treats ppin as const. Why do we need asn1_root_cert_ to begin with?

Reply:
hi, @yelvmw -- The only diff in this function is the change of the input arg from `string &      asn1_root_cert` to `const string &asn1_root_cert`, as this input argument is read-only.

No, there is no  bug in this code prior to this change. Old code use the input arg whereas the new code is using, effectively, the same param after copying input param to `asn1_root_cert_` member field, on this L :

`2402   asn1_root_cert_.assign((char *)asn1_root_cert.data(), asn1_root_cert.size());`

Your question seems fair. This class member-field `asn1_root_cert_` seems to be used mainly for copying stuff around. 

I dug around and it does seem likely that we could eliminate this field, but I don't want to change unrelated stuff under this change-set. When / if the larger class layout reorgs occur, we should examine and rework the need for such fields.